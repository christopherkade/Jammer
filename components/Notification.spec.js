import { shallowMount, config } from '@vue/test-utils'
import cmp from './Notification.vue'

describe('[UNIT] Testing the Notification component', () => {
  let wrapper

  config.mocks.$store = {
    state: {
      notification: {
        isShown: true,
        type: 'is-danger',
        message: 'notificationMock'
      }
    },
    mutations: {
      setIsShown: jest.fn()
    }
  }

  beforeEach(() => {
    wrapper = shallowMount(cmp)
  })

  it('Snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component validation', () => {
    it('should render the main (parent) component', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })
  })

  describe('Content validation', () => {
    it('should render a notification message', () => {
      expect(wrapper.text()).toContain('notificationMock')
    })

    it('should render a notification with the correct style', () => {
      expect(wrapper.find('.message').attributes('class')).toContain('is-danger')
    })
  })
})
