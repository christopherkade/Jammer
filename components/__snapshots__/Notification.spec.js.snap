// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] Testing the Notification component Snapshot testing 1`] = `
"<article class=\\"message notification is-danger\\"><button aria-label=\\"delete\\" class=\\"delete\\"></button> <div class=\\"message-body\\">
    notificationMock
  </div></article>"
`;
